packages/liveblocks-react/src/__tests__/useThreads.test.tsx

```diff
@@ -492,6 +521,179 @@ describe("useThreads", () => {
 
     unmount();
   });
+
+  test("should sort threads by creation date before returning", async () => {
+    const oldThread = dummyThreadData();
+    oldThread.createdAt = new Date("2021-01-01T00:00:00Z");
+
+    const newThread = dummyThreadData();
+    newThread.createdAt = new Date("2021-01-02T00:00:00Z");
+
+    server.use(
+      mockGetThreads(async (_req, res, ctx) => {
+        return res(
+          ctx.json({
+            data: [newThread, oldThread], // The order is intentionally reversed to test if the hook sorts the threads by creation date
+            inboxNotifications: [],
+          })
+        );
+      })
+    );
+
+    const {
+      roomCtx: { RoomProvider, useThreads },
+    } = createRoomContextForTest();
+
+    const { result, unmount } = renderHook(
+      () => ({
+        threads: useThreads(),
+      }),
+      {
+        wrapper: ({ children }) => (
+          <RoomProvider id="room-id" initialPresence={{}}>
+            {children}
+          </RoomProvider>
+        ),
+      }
+    );
+
+    expect(result.current.threads).toEqual({ isLoading: true });
+
+    await waitFor(() =>
+      expect(result.current.threads).toEqual({
+        isLoading: false,
+        threads: [oldThread, newThread],
+      })
+    );
+
+    unmount();
+  });
+
+  test("should sort threads by creation date before returning (when GET THREADS resolves before GET INBOX NOTIFICATIONS request)", async () => {
+    const oldThread = dummyThreadData();
+    oldThread.createdAt = new Date("2021-01-01T00:00:00Z");
+
+    const newThread = dummyThreadData();
+    newThread.createdAt = new Date("2021-01-02T00:00:00Z");
+
+    const inboxNotification = dummyInboxNoficationData();
+    inboxNotification.threadId = oldThread.id;
+
+    server.use(
+      mockGetThreads(async (_req, res, ctx) => {
+        return res(
+          ctx.json({
+            data: [newThread],
+            inboxNotifications: [],
+          })
+        );
+      }),
+      mockGetInboxNotifications(async (_req, res, ctx) => {
+        // Mock a delay in response so that GET THREADS request is resolved before GET NOTIFICATIONS request
+        await wait(100);
```

Comment:
Could you use jest fake timers instead? We're trying to avoid waiting arbitrary values that could create flacky tests on slower machines.

Reply:
Good catch! Updated to use jest fake timers!